### PR TITLE
Image classification new api markdown

### PR DESCRIPTION
**Pull request by jetfuel**
Update the English and Chinese Image Classification tutorial with high level fluid API

### CODE REVIEW

File: 03.image_classification/README.cn.md [hunk 1]
```diff
@@ -403,66 +378,76 @@ cost_ploter = Ploter(train_title, test_title)
 step = 0
 def event_handler_plot(event):
     global step
-    if isinstance(event, paddle.event.EndIteration):
+    if isinstance(event, fluid.EndStepEvent):
         if step % 1 == 0:
```
Comment (daming-lu) [inline 03.image_classification/README.cn.md:382]: we should change `step % 1 == 0` to something more meaningful, say `step % 10`

File: 03.image_classification/README.cn.md [hunk 2]
```diff
@@ -403,66 +378,76 @@ cost_ploter = Ploter(train_title, test_title)
...
-            cost_ploter.append(train_title, step, event.cost)
+            cost_ploter.append(train_title, step, event.metrics[0])
             cost_ploter.plot()
         step += 1
-    if isinstance(event, paddle.event.EndPass):
+    if isinstance(event, fluid.EndEpochEvent):
```
Comment (daming-lu) [inline 03.image_classification/README.cn.md:386]: just curious at the last step of an epoch, will Fluid trigger both `EndStepEvent` and `EndEpochEvent` ?

File: 03.image_classification/README.cn.md [hunk 3]
```diff
@@ -474,39 +459,55 @@ Test with Pass 0, {'classification_error_evaluator': 0.885200023651123}
...
+
     im = np.array(im).astype(np.float32)
-    # PIL打开图片存储顺序为H(高度)，W(宽度)，C(通道)。
-    # PaddlePaddle要求数据顺序为CHW，所以需要转换顺序。
-    im = im.transpose((2, 0, 1)) # CHW
-    # CIFAR训练图片通道顺序为B(蓝),G(绿),R(红),
-    # 而PIL打开图片默认通道顺序为RGB,因为需要交换通道。
-    im = im[(2, 1, 0),:,:] # BGR
-    im = im.flatten()
+    # The storage order of the loaded image is W(widht),
```
Comment (daming-lu) [inline 03.image_classification/README.cn.md]: W(width)

File: 03.image_classification/README.cn.md [hunk 4]
```diff
@@ -474,39 +459,55 @@ Test with Pass 0, {'classification_error_evaluator': 0.885200023651123}
...
+    # H(height), C(channel). PaddlePaddle requires
+    # the CHW order, so transpose them.
+    im = im.transpose((2, 0, 1))  # CHW
+    # In the training phase, the channel order of CIFAR
+    # image is B(Blue), G(green), R(Red). But PIL open
```
Comment (daming-lu) [inline 03.image_classification/README.cn.md]: PIL opens

File: 03.image_classification/README.cn.md [hunk 5]
```diff
@@ -474,39 +459,55 @@ Test with Pass 0, {'classification_error_evaluator': 0.885200023651123}
...
-# with open('params_pass_50.tar', 'r') as f:
-#    parameters = paddle.parameters.Parameters.from_tar(f)
+### Inferencer 配置和预测
 
-probs = paddle.infer(
-    output_layer=out, parameters=parameters, input=test_data)
-lab = np.argsort(-probs) # probs and lab are the results of one batch data
-print "Label of image/dog.png is: %d" % lab[0][0]
-```
+`Inferencer` 需要一个 `infer_func` 和 `param_path` 来设置网络和经过训练的参数。
```
Comment (daming-lu) [inline 03.image_classification/README.cn.md:495]: kind of weird that we call the param name `param_path` but use `params_dirname` as the variable name 😭 

`param_path=params_dirname`

But let's keep it as-is for now.

File: 03.image_classification/README.md [hunk 6]
```diff
@@ -329,78 +283,105 @@ def resnet_cifar10(ipt, depth=32):
...
 
-### Define Parameters
+## Train Program Configuration
+Then we need to setup the the `train_program`. It takes the prediction from the inference_program first.
+During the training, it will calculate the `avg_loss` from the prediction.
 
-Firstly, we create the model parameters according to the previous model configuration `cost`.
+In the context of supervised learning, labels of training images are defined in `fluid.layers.data` as well. During training, the cross-entropy loss function is used and the loss is the output of the network. During testing, the outputs are the probabilities calculated in the classifier.
+
+**NOTE:** A train program should return an array and the first return argument has to be `avg_cost`.
```
Comment (daming-lu) [inline 03.image_classification/README.md]: first returned ; The trainer always implicitly uses

File: 03.image_classification/README.md [hunk 7]
```diff
@@ -410,65 +391,77 @@ cost_ploter = Ploter(train_title, test_title)
 step = 0
 def event_handler_plot(event):
     global step
-    if isinstance(event, paddle.event.EndIteration):
+    if isinstance(event, fluid.EndStepEvent):
         if step % 1 == 0:
```
Comment (daming-lu) [inline 03.image_classification/README.md:395]: same here. `step %10` might make more sense 😆

File: 03.image_classification/README.md [hunk 8]
```diff
@@ -478,42 +471,56 @@ Figure 12. The error rate of VGG model on CIFAR10
...
 
-
 ## Application
 
-After training is completed, users can use the trained model to classify images. The following code shows how to infer through `paddle.infer` interface. You can uncomment some lines from below to change the model name.
+After training is completed, users can use the trained model to classify images. The following code shows how to infer through `fluid.Inferencer` interface. You can uncomment some lines from below to change the model name.
+
+### Generate input data for inferring
+
+`dog.png` is an example image of a dog. Turn it into an numpy array to match the data feeder format.
```
Comment (daming-lu) [inline 03.image_classification/README.md]: `a` numpy array

File: 03.image_classification/README.md [hunk 9]
```diff
@@ -478,42 +471,56 @@ Figure 12. The error rate of VGG model on CIFAR10
...
 from PIL import Image
 import numpy as np
 import os
+
 def load_image(file):
     im = Image.open(file)
     im = im.resize((32, 32), Image.ANTIALIAS)
+
     im = np.array(im).astype(np.float32)
     # The storage order of the loaded image is W(widht),
```
Comment (daming-lu) [inline 03.image_classification/README.md]: width

**Review by daming-lu — APPROVED** [review]
some tiny comments 😬

File: 03.image_classification/README.cn.md (hunk 2 above)
Comment (jetfuel) [reply 03.image_classification/README.cn.md:386]: I believe the order is 
EndStepEvent, and then EndEpochEvent

File: 03.image_classification/README.cn.md (hunk 1 above)
Comment (jetfuel) [reply 03.image_classification/README.cn.md:382]: How about we remove this line? Looks like we should draw it no matter what.
